src/drawers/canvasDrawer.ts

```diff
@@ -51,3 +52,41 @@ export class CanvasDrawer implements IDrawer {
     // NO op - canvas element owns the canvas; context is free
   }
 }
+
+export const ContextStyleAttrs = {
+  strokeWidth: "stroke-width", stroke: "stroke", opacity: "opacity", fill: "fill",
+};
+
+export function resolveAttributesSubsetWithStyles(projector: AttributeToAppliedProjector, extraKeys: string[], datum: any, index: number) {
+  const attrKeys = Object.keys(ContextStyleAttrs).concat(extraKeys);
+  const attrs: {[key: string]: any} = {};
```

Comment:
You can actually get statically typed attrs key objects using Typescript's mapped types: see http://www.typescriptlang.org/play/#src=const%20ContextStyleAttrs%20%3D%20%7B%0A%20%20strokeWidth%3A%20%22stroke-width%22%2C%20stroke%3A%20%22stroke%22%2C%20opacity%3A%20%22opacity%22%2C%20fill%3A%20%22fill%22%2C%0A%7D%3B%0A%0Afunction%20resolveAttributesSubsetWithStyles%3CExtraKeyLiterals%20extends%20string%3E(extraKeys%3A%20ExtraKeyLiterals%5B%5D%2C%20datum%3A%20any%2C%20index%3A%20number)%20%7B%0A%20%20type%20AttributeNames%20%3D%20(keyof%20typeof%20ContextStyleAttrs)%20%7C%20ExtraKeyLiterals%3B%0A%20%20const%20attrKeys%20%3D%0A%20%20%20%20Object.keys(ContextStyleAttrs).concat(extraKeys)%20as%20AttributeNames%5B%5D%3B%0A%20%20const%20attrs%3A%20%7B%20%5BK%20in%20AttributeNames%5D%3F%3A%20any%20%7D%20%3D%20%7B%7D%3B%0A%20%20return%20attrs%3B%0A%7D%0A%0Avar%20k%20%3D%20resolveAttributesSubsetWithStyles(%5B%22x%22%2C%20%22y%22%2C%20%22width%22%2C%20%22height%22%5D%2C%20%7B%7D%2C%200)%3B